test/unit/math/opencl/device_functions/digamma_test.cpp

```diff
@@ -0,0 +1,32 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <stan/math/opencl/device_functions/digamma.hpp>
+#include <stan/math/opencl/kernel_cl.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/expect_near_rel.hpp>
+#include <string>
+
+static const std::string test_kernel_code
+    = STRINGIFY(__kernel void test(__global double *B, __global double *A) {
+        const int i = get_global_id(0);
+        B[i] = digamma(A[i]);
+      });
+
+const stan::math::opencl_kernels::kernel_cl<
+    stan::math::opencl_kernels::out_buffer,
+    stan::math::opencl_kernels::in_buffer>
+    digamma("test", {stan::math::opencl_kernels::digamma_device_function,
+                     test_kernel_code});
```

Comment:
This is also a really nice show case of how easy it is to add a new OpenCL kernel in Stan Math.